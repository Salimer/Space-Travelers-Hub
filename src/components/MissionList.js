import PropTypes from 'prop-types';
import Table from 'react-bootstrap/Table';
import MissionItem from './MissionItem';

const MissionList = (props) => {
  const { missions } = props;

  return (
    <div>
      <Table striped bordered>
        <thead>
          <tr>
            <th>Mission</th>
            <th>Description</th>
            <th>Status</th>
            <th>{'   '}</th>
          </tr>
        </thead>
        <tbody>
          {missions.map((mission) => (
            <MissionItem
              key={mission.mission_id}
              id={mission.mission_id}
              name={mission.mission_name}
              description={mission.description}
              reserved={mission.reserved}
            />
          ))}
        </tbody>
      </Table>
    </div>
  );
};

MissionList.propTypes = {
  missions: PropTypes.arrayOf.isRequired,
};

export default MissionList;
