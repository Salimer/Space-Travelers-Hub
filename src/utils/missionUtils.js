const filterReserved = (missions) => missions.filter((mission) => mission.reserved);

export default filterReserved;
