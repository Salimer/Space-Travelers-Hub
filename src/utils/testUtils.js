import { BrowserRouter } from 'react-router-dom';
import { Provider } from 'react-redux';
import Table from 'react-bootstrap/Table';
import store from '../redux/store';

const withRouter = (component) => (
  <BrowserRouter>
    {component}
  </BrowserRouter>
);

const withProvider = (component, mockStore = store) => (
  <Provider store={mockStore}>
    {component}
  </Provider>
);

const withTable = (component) => (
  <Table>
    <tbody>
      {component}
    </tbody>
  </Table>
);

export { withProvider, withRouter, withTable };
