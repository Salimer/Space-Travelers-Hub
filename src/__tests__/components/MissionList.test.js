import renderer from 'react-test-renderer';
import configureStore from 'redux-mock-store';
import { withProvider } from '../../utils/testUtils';
import MissionList from '../../components/MissionList';

const mockStore = configureStore([]);

describe('testing MissionList component', () => {
  test('render', () => {
    const missions = [
      {
        mission_id: '1',
        mission_name: 'mission 1',
        description: 'mission 1 description',
        reserved: false,
      },
      {
        mission_id: '2',
        mission_name: 'mission 2',
        description: 'mission 2 description',
        reserved: true,
      },
    ];
    const store = mockStore({ missions });

    const tree = renderer.create(withProvider(<MissionList missions={missions} />, store));
    expect(tree).toMatchSnapshot();
  });
});
