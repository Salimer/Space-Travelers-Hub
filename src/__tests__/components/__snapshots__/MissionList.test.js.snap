// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`testing MissionList component render 1`] = `
<div>
  <table
    className="table table-striped table-bordered"
    style={
      Object {
        "fontSize": "0.9rem",
      }
    }
  >
    <thead
      style={
        Object {
          "fontSize": "1rem",
        }
      }
    >
      <tr>
        <th>
          Mission
        </th>
        <th>
          Description
        </th>
        <th>
          Status
        </th>
        <th>
             
        </th>
      </tr>
    </thead>
    <tbody>
      <tr>
        <td
          style={
            Object {
              "fontWeight": "bold",
            }
          }
        >
          mission 1
        </td>
        <td>
          mission 1 description
        </td>
        <td
          className="align-middle"
          style={
            Object {
              "textAlign": "center",
            }
          }
        >
          <span
            className="badge bg-secondary"
          >
            NOT A MEMBER
          </span>
        </td>
        <td
          className="align-middle"
          style={
            Object {
              "textAlign": "center",
              "width": "11rem",
            }
          }
        >
          <button
            className="btn btn-outline-dark"
            data-testid="mission-1-button"
            disabled={false}
            onClick={[Function]}
            type="button"
          >
            Join mission
          </button>
        </td>
      </tr>
      <tr>
        <td
          style={
            Object {
              "fontWeight": "bold",
            }
          }
        >
          mission 2
        </td>
        <td>
          mission 2 description
        </td>
        <td
          className="align-middle"
          style={
            Object {
              "textAlign": "center",
            }
          }
        >
          <span
            className="badge bg-success"
          >
            Active Member
          </span>
        </td>
        <td
          className="align-middle"
          style={
            Object {
              "textAlign": "center",
              "width": "11rem",
            }
          }
        >
          <button
            className="btn btn-outline-danger"
            data-testid="mission-2-button"
            disabled={false}
            onClick={[Function]}
            type="button"
          >
            Leave mission
          </button>
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;
